drivers/sensor/bosch/bmm350/bmm350.c

```diff
@@ -0,0 +1,984 @@
+	tcs_y = (data->otp_data[BMM350_MAG_TCS_Y] & BMM350_MSB_MASK) >> 8;
+	tcs_z = (data->otp_data[BMM350_MAG_TCS_Z] & BMM350_MSB_MASK) >> 8;
+
+#ifdef BMM350_USE_FIXED_POINT
+	data->mag_comp.dut_tcs.tcs_x =
+		fix_sign(tcs_x, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+	data->mag_comp.dut_tcs.tcs_y =
+		fix_sign(tcs_y, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+	data->mag_comp.dut_tcs.tcs_z =
+		fix_sign(tcs_z, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+
+	data->mag_comp.dut_tcs.tcs_x = (data->mag_comp.dut_tcs.tcs_x / 16384);
+	data->mag_comp.dut_tcs.tcs_y = (data->mag_comp.dut_tcs.tcs_y / 16384);
+	data->mag_comp.dut_tcs.tcs_z = (data->mag_comp.dut_tcs.tcs_z / 16384);
+#endif
+
+#ifdef BMM350_USE_FIXED_POINT
+	data->mag_comp.dut_t0 =
+		(fix_sign(data->otp_data[BMM350_MAG_DUT_T_0], BMM350_SIGNED_16_BIT) / 512) + 23;
+#endif
+
+	cross_x_y = (data->otp_data[BMM350_CROSS_X_Y] & BMM350_LSB_MASK);
+	cross_y_x = (data->otp_data[BMM350_CROSS_Y_X] & BMM350_MSB_MASK) >> 8;
+	cross_z_x = (data->otp_data[BMM350_CROSS_Z_X] & BMM350_LSB_MASK);
+	cross_z_y = (data->otp_data[BMM350_CROSS_Z_Y] & BMM350_MSB_MASK) >> 8;
+
+#ifdef BMM350_USE_FIXED_POINT
+	data->mag_comp.cross_axis.cross_x_y =
+		fix_sign(cross_x_y, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+	data->mag_comp.cross_axis.cross_y_x =
+		fix_sign(cross_y_x, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+	data->mag_comp.cross_axis.cross_z_x =
+		fix_sign(cross_z_x, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+	data->mag_comp.cross_axis.cross_z_y =
+		fix_sign(cross_z_y, BMM350_SIGNED_8_BIT) * BMM350_MAG_COMP_COEFF_SCALING;
+
+	data->mag_comp.cross_axis.cross_x_y = (data->mag_comp.cross_axis.cross_x_y / 800);
+	data->mag_comp.cross_axis.cross_y_x = (data->mag_comp.cross_axis.cross_y_x / 800);
+	data->mag_comp.cross_axis.cross_z_x = (data->mag_comp.cross_axis.cross_z_x / 800);
+	data->mag_comp.cross_axis.cross_z_y = (data->mag_comp.cross_axis.cross_z_y / 800);
+#endif
+}
+
+static int bmm350_otp_dump_after_boot(const struct device *dev)
+{
+	struct bmm350_data *data = dev->data;
+	int ret = 0;
+	uint16_t otp_word = 0;
+
+	for (uint8_t idx = 0; idx < BMM350_OTP_DATA_LENGTH; idx++) {
+		ret = bmm350_read_otp_word(dev, idx, &otp_word);
+		data->otp_data[idx] = otp_word;
+	}
+
+	data->var_id = (data->otp_data[30] & 0x7f00) >> 9;
+	/* Set the default auto bit reset configuration */
+	data->enable_auto_br = ((data->var_id > BMM350_CURRENT_SHUTTLE_VARIANT_ID) ? BMM350_DISABLE
+										   : BMM350_ENABLE);
+
+	LOG_DBG("bmm350 Find the var id %d\n", data->var_id);
+	/* Update magnetometer offset and sensitivity data. */
+	bmm350_update_mag_off_sens(data);
+
+	if (ret) {
+		LOG_ERR("i2c xfer failed, ret = %d\n", ret);
+	}
+
+	return ret;
+}
+
+/*!
+ * @brief This API gets the PMU command status 0 value
+ */
+static int8_t bmm350_get_pmu_cmd_status_0(struct bmm350_pmu_cmd_status_0 *pmu_cmd_stat_0,
+					  const struct device *dev)
```

Comment:
nit: usually, I always see the `const struct device *` as the first arg for every function. Just something that seems to not follow a pattern